python/tests/estimators/test_keras_estimators.py

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+import os
+import shutil
+import tempfile
+
+import PIL.Image
+import numpy as np
+from keras.applications import Xception
+from keras.applications.imagenet_utils import preprocess_input
+
+from sparkdl.estimators.keras_image_file_estimator import KerasImageFileEstimator
+
+from ..tests import SparkDLTestCase
+from ..transformers.image_utils import getSampleImagePaths
+
+def _load_image_from_uri(local_uri):
+    img = (PIL.Image
+           .open(local_uri)
+           .convert('RGB')
+           .resize((299, 299), PIL.Image.ANTIALIAS))
+    img_arr = np.array(img).astype(np.float32)
+    img_tnsr = preprocess_input(img_arr[np.newaxis, :])
+    return img_tnsr
+
+class KerasEstimatorsTest(SparkDLTestCase):
+
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+        self.model_filename = os.path.join(self.temp_dir, 'model.h5')
+
+        self.estimator = KerasImageFileEstimator(inputCol='imageUri', 
+                                                 outputCol='preds',
+                                                 labelCol='label',
+                                                 imageLoader=_load_image_from_uri,
+                                                 optimizer='adam',
+                                                 loss='categorical_crossentropy',
+                                                 kerasParams={},
+                                                 modelFile=self.model_filename)
+        
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir, ignore_errors=True)
+        
+    def test_valid_params(self):
+        self.assertTrue(self.estimator._validateParams())
+
+    def test_valid_workflow(self):        
+        # Saving a model
+        model = Xception(weights=None)
+        model.save(self.model_filename)
+
+        # Create image URI dataframe
+        image_uris = getSampleImagePaths()
+        image_labels = [1] * len(image_uris)
+        image_uri_df = self.sc.parallelize(
+            zip(image_uris, image_labels)).toDF(['imageUri', 'label'])
+        
+        transformers = self.estimator.fit(image_uri_df)
+        self.assertTrue(1 == len(transformers))
```

Comment:
`self.assertEqual` is a little more informative than `assertTrue` when it fails.